### PR TITLE
Reveal deactivate account option to all

### PR DESCRIPTION
**Pull request by HSalmon**
Addresses: https://github.com/loomio/loomio/issues/3629
deactivate-profile-1

### CODE REVIEW

File: angular/core/components/only_coordinator_modal/only_coordinator_modal.coffee [hunk 1]
```diff
@@ -0,0 +1,12 @@
+angular.module('loomioApp').factory 'OnlyCoordinatorModal', ->
+  templateUrl: 'generated/components/only_coordinator_modal/only_coordinator_modal.html'
+  controller: ($scope, $location, Session, LmoUrlService) ->
+
+    $scope.groups = ->
+      _.filter Session.user().groups(), (group) ->
+        _.contains(group.adminIds(), Session.user().id) and
```
Comment (gdpelican) [inline angular/core/components/only_coordinator_modal/only_coordinator_modal.coffee]: We should move this into AbilityService I think.

```
isOnlyCoordinator (group) ->
  @canAdministerGroup(group) and !group.hasMultipleAdmins
```

**Comment by gdpelican** [comment]
I'd really rather have this information show up on the original deactivate modal, rather than going through the mental gymnastics of clicking 'Yes, I want to deactivate' only to be told I can't.

Can we say something like:

```
When you deactivate your account:
- stuff
- happens

You are currently unable to deactivate because you a coordinator etc.
```

in a single modal, with a disabled 'Yes I want to deactivate' button?

**Comment by HSalmon** [comment]
@gdpelican, I was just building this according to the design. @rdbartlett, what do you think about James' suggestion above?

**Comment by rdbartlett** [comment]
I'm happy with the original design, and I'm happy with James's suggestion too. Having a bit of friction in the deactivation process seems like a good thing.

On the original issue I was intending to open a conversation, rather than jumping straight to building something. Specifically I was trying to figure out what's changed.

**Comment by rdbartlett** [comment]
Ideally we would have had the discussion before the building. Now that it's built to match the design though, I reckon it is good enough to merge.

We can add another ticket to the inbox if we want to do another iteration

**Comment by HSalmon** [comment]
Thanks @rdbartlett, I do wonder whether this could do with a smoke-test though...

**Comment by HSalmon** [comment]
@gdpelican, @rdbartlett, @mhjb, do any of you have capacity to smoke-test this branch?

**Comment by rdbartlett** [comment]
Can you get it up on a test env for me to play with?

**Comment by mhjb** [comment]
ditto for me

**Comment by mhjb** [comment]
It works properly but gives an odd message at the end:

image

This might be out of scope, but why do we remove people's names from people's contributions when they deactivate their account? Scenario: I'm running a board of directors on Loomio. A past member contributed in various ways including voting on important proposals, but then their term was up and they moved on. They weren't using Loomio in any other context so deactivated their account. Why should their name be removed from the record of our past important decisions?

**Comment by rdbartlett** [comment]
Tested on Chrome:

With multiple groups, confirmed I am told to close them before I can deactivate. Confirmed theg groups link to the right place.

With no groups, confirmed I can deactivate the account.

I think the flash text should change from "Your account is not activated yet." to "There is a deactivated account associated with this email address. If you want to reactivate your account, [contact us](#)."

@mhjb I'd say your question is definitely out of scope for this job, and interesting to discuss. The rationale for making that choice was to protect the privacy of people who have commented in public threads. 

(Process wise I think it is really useful to distinguish blocking feedback from non-blocking. None of my feedback is blocking and I'm happy to see this deployed as is, or preferably with an updated flash text.)

**Comment by HSalmon** [comment]
Thanks very much for testing @mhjb and @rdbartlett, I'll update the flash message.

**Comment by HSalmon** [comment]
screen shot 2016-08-19 at 3 45 26 pm

**Comment by HSalmon** [comment]
I think it's ready to go! @rdbartlett, @mhjb, @robguthrie (do you green apple it, Rob)?

**Comment by mhjb** [comment]
Sorry — I should have been more specific in my feedback. The new message is great, but I think it's odd to see it immediately upon deactivating. I wonder if when you successfully deactivate your account, you should be sent to the home page, not a login page.

**Comment by rdbartlett** [comment]
Matt this is built into Devise, I think it is too much effort to change
anything more than the words.

**Comment by mhjb** [comment]
Sorry which bit is built into Devise?

**Comment by rdbartlett** [comment]
The behaviour that says the flash message you see on deactivating is the
same as when you try to login with a deactivated account. We get a fair
number of support requests from people who are trying to log in with
deactivated accounts so this feels like an improvement.

**Comment by mhjb** [comment]
Yeah but I was suggesting upon deactivation we send people to www.loomio.org, rather than to https://www.loomio.org/users/sign_in — is that behaviour built into Devise too?

**Comment by HSalmon** [comment]
@robguthrie, any chance I could get a green apple on this if you're all good with it?

**Comment by robguthrie** [comment]
🍏 !!!!!

**Pull request merged**